src/Workspaces/Core/Portable/Workspace/Solution/ProjectDependencyGraph.cs

```diff
@@ -14,35 +14,288 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     public class ProjectDependencyGraph
     {
-        private readonly ImmutableArray<ProjectId> _projectIds;
+        private readonly ImmutableHashSet<ProjectId> _projectIds;
         private readonly ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> _referencesMap;
```

Comment:
i feel like i want this to eb called _directReferencesMap.  That way i have a clear way of knowing with all these fields if things are 'direct' or 'transitive'.

Reply:
This would be great.

Reply:
Going to won't fix this but won't complain about a follow up PR. Want to avoid even bigger diff churn for now.